pkg/utils/timeutils.go

```diff
@@ -96,11 +105,14 @@ func Str2unixTime(timeString string) (int64, error) {
 		return int64(tint), nil
 	}
 
-	t, err := time.Parse(time.RFC3339, timeString)
-	if err != nil {
-		return 0, errors.Wrap(err, "Not an RFC 3339 time format")
+	for _, format := range TimeFormats {
+		t, err := time.Parse(format, timeString)
```

Comment:
This could potentially slow things, especially in the worst case (data is in the last format).

Reply:
I've benchmarked the new code vs the original version. On my machine it adds 2ns per operation.

Any faster way to do this? In other packages (e.g pandas) there's an optional `format` parameter to the parser function. However didn't want to introduce to many changes.

Reply:
Did you benchmark the worst case (RFC3339Nano input)? What if more formats are added to the list? (what of the worst case then?)

Reply:
Since I don't know the motivation for supporting this change, it's hard to weigh against the potential performance downside.

It does look like this code is only used on the hot path of the Nuclio examples and benchmark, not core production code.

Reply:
> Did you benchmark the worst case (RFC3339Nano input)?
No. I did a `RFC3339` format string since this is the only format the current system uses. If I test the old with `RFC3999` and the new with `RFC3999Nano` then it's about 40ns more.

> What if more formats are added to the list? (what of the worst case then?)
Still about 40ns.  Attaching the benchmark code.

Reply:
> Since I don't know the motivation for supporting this change, it's hard to weigh against the potential performance downside.

Working with @yaronha on https://github.com/v3io/frames, we'd like more than a second granularity on times in the database.

Reply:
Ok, by my own measurements, on my laptop, using a nano timestamp, it takes 351 ns to parse RFC3339, and 390 ns for RFC3339Nano. So 741 ns combined. I guess that's acceptable if you guys actually need it (and it's not e.g. for testing purposes, or something like that).